### DISCUSSION TITLE
Is it possible for server parser to update the url with default query?

### DISCUSSION THREAD

**Discussion by damian-balas** [47ng/nuqs#507] [Q&A] [answered]
Hi :)
Is it possible to update the url with default value?

```ts
const getDefaultArticleType = ({
  hasPodcast,
  hasVideo,
  hasArticle,
}: {
  hasPodcast: boolean;
  hasVideo: boolean;
  hasArticle: boolean;
}) => {
  if (hasArticle) return ARTICLE_TYPE.ARTICLE;
  if (hasVideo) return ARTICLE_TYPE.VIDEO;
  if (hasPodcast) return ARTICLE_TYPE.PODCAST;

  return null;
};


const defaultArticleType = getDefaultArticleType({
    hasPodcast: !!data?.attributes.audio,
    hasVideo: !!data?.attributes.video,
    hasArticle: !!data?.attributes.text,
  });
 
 if (!defaultArticleType) notFound();
  
const articleTypeParser = parseAsStringEnum<ARTICLE_TYPE>(
    Object.values(ARTICLE_TYPE),
  ).withDefault(defaultArticleType);

  const articleType = articleTypeParser.parseServerSide(
    searchParams.type?.toLowerCase(),
  );
```

Sometimes `searchParams.type` is undefined and the default value kicks in but the url doesn't change. Everything works fine but I want the url to **always** include the query.

example:
`video` is the default value.

if user goes to this url: `/articles` I want to change the url to `/articles?type=video` 

I am using the `app router` and server components. I'm not even sure if this should me handled by `nuqs`. Please give me your opinion on how to handle this.

Thanks :)

**Comment by franky47** [comment, marked as answer]
If you want to update the URL sever-side, this sounds like a job for a [middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware).

> I'm not even sure if this should me handled by `nuqs`

Indeed. However `nuqs` can help you build that URL, with the [serializer helper](https://nuqs.47ng.com/docs/utilities#serializer-helper).

**Reply by damian-balas** [reply]
All clear :) Thanks
